part1/src/main.rs

```diff
@@ -44,7 +49,7 @@ impl Interpreter {
             return;
         }
 
-        panic!("Wrong Character");
+        panic!("Wrong Character: {}", current_char);
```

Comment:
this "prevents" a+b?

Reply:
but it doesn't prevent a++?

Reply:
It seems to catch that too:
```
➤ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/lsbasi-rs`
1++
position: 3; text_length: 3
thread 'main' panicked at 'Value not integer: OPERATION(
    PLUS
)', src/main.rs:84:9
…
```

I'd encourage you to write these all up as tests, then you can refactor with abandon and not worry about regressing your functionality.